File: server/internal/models/task.go
package models

import "time"

type Task struct {
	ID          int       `json:"id"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	Done        bool      `json:"done"`
	UserID      int       `json:"userId"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

type CreateTaskRequest struct {
	Title       string `json:"title"`
	Description string `json:"description"`
	Done        bool   `json:"done"`
}

// UpdateTaskRequest allows partial updates of a task. Fields are pointers
// so we can distinguish between "not provided" and zero values.
type UpdateTaskRequest struct {
	Title       *string `json:"title,omitempty"`
	Description *string `json:"description,omitempty"`
	Done        *bool   `json:"done,omitempty"`
}

File: client/src/types/task.ts
export type Task = {
  id: number
  title: string
  description?: string
  done: boolean
  userId: number
  createdAt: string
}

export type CreateTaskRequest = {
  title: string
  description?: string
}

export type UpdateTaskRequest = {
  title?: string
  description?: string
  done?: boolean
}
